src/Domain/SegmentId.hpp

```diff
@@ -99,6 +108,25 @@ inline SegmentId SegmentId::id_of_child(Side side) const noexcept {
   return {refinement_level_ + 1, 1 + index_ * 2};
 }
 
+inline SegmentId SegmentId::id_of_sibling() const noexcept {
+  ASSERT(0 != refinement_level_,
+         "The segment on the root refinement level has no sibling");
+  return {refinement_level_, (0 == index_ % 2 ? index_ + 1 : index_ - 1)};
+}
+
+inline SegmentId SegmentId::id_of_abutting_nibling() const noexcept {
+  ASSERT(0 != refinement_level_,
+         "The segment on the root refinement level has no abutting nibling");
+  return {refinement_level_ + 1,
+          (0 == index_ % 2 ? 2 * index_ + 2 : 2 * index_ - 1)};
```

Comment:
future: diagram with numbers

Reply:
my hand-drawn diagram had the refinement levels and indices of segments labeled (and described in the documentation)